### PR TITLE
Removes pipe construction delay on the RPD

### PR DESCRIPTION
**Pull request by jakmak5**
remake of https://github.com/d3athrow/vgstation13/pull/13583 because git

This removes that 5 second delay that the RPD has when building pipes so now they are constructed instantly. Works with borg and carbon RPD's. I dont even know why it was added in the first place since all it does is make construction unnecessarily tedious and slows down autism projects.

NOTE i kept the delay in for disposals pipes to stop hallway blockage

### CODE REVIEW

**Comment by PJB3005** [comment]
Powercreep.

Lots of it.

**Comment by BobdaBiscuit** [comment]
Fund it.

**Comment by SonixApache** [comment]
I'm okay with this.

**Comment by jknpj** [comment]
Direct buff to pipe-powered SME

**Comment by Kurfursten** [comment]
You can still spam out disposal bin objects to clog up hallways. This is at least partially limited now by the build time, but with no delay, you will be thankful that an RPD is considered contraband, because the clown can do SO MUCH MORE DAMAGE than lube.

**Comment by Aurx** [comment]
>You can still spam out disposal bin objects to clog up hallways. This is at least partially limited now by the build time, but with no delay, you will be thankful that an RPD is considered contraband, because the clown can do SO MUCH MORE DAMAGE than lube.

All pipeybits that block movement NEED a delay.

File: code/modules/RCD/schematics/pipe.dm [hunk 1]
```diff
@@ -382,8 +382,8 @@
 /datum/rcd_schematic/pipe/disposal/attack(var/atom/A, var/mob/user)
 	to_chat(user, "Building Pipes ...")
 	playsound(get_turf(user), 'sound/machines/click.ogg', 50, 1)
-	if(!do_after(user, A, 20))
-		return 1
+//	if(!do_after(user, A, 1))
+//		return 1
```
Comment (ghost) [inline code/modules/RCD/schematics/pipe.dm]: Oh god, why remove the return 1?

Comment (jakmak) [reply code/modules/RCD/schematics/pipe.dm]: It wouldn't compile if I didn't remove it

Comment (ghost) [reply code/modules/RCD/schematics/pipe.dm]: Because you'd need to tab it back one. The language things tabs = { }

**Comment by jakmak** [comment]
When I get home I will re add and double the construction time of disposal pipe parts due to the threat that they can block the hallway

File: code/modules/RCD/schematics/pipe.dm (hunk 1 above)
Comment (PJB3005) [reply code/modules/RCD/schematics/pipe.dm]: What @Skullyton, the do after is gone entirely so there's no point returning.

Comment (ghost) [reply code/modules/RCD/schematics/pipe.dm]: You need a return 1 on attack so it registers as a succesful attack, so some other sanity is ran IIRC

Comment (PJB3005) [reply code/modules/RCD/schematics/pipe.dm]: @Skullyton this is the modular RCD framework. In the schematic activation procs, `1` means nondescriptive error, `string` means descriptive error, and `0/null` means everything was fine.

**Comment by Probe1** [comment]
I'd rather we don't balance the game around being annoyed one time.

**Comment by jakmak** [comment]
So wait should I re add the delay for disposal pipe parts or not? If I remove it then an ultra obscure form off Griff becomes slightly easier but if I keep the delay in autism disposal systems stay soul crushingly tedious to do since you still need to wrench+weld the pipe part. May not seem like much for a simple repair job but could easily double construction time of a custom disposals setup. Hell I created this change just to remove the artificial time extension to making piping setups.

**Comment by clusterfack** [comment]
I think there are better ways to improve the QoL of this object without doing this in particular.

**Comment by Intigracy** [comment]
It doesn't need to be doubled but it should be re-added, yes.

**Comment by jakmak5** [comment]
OK i re added the pipe construction delay for disposal pipes but not for normal pipes/misc pipe related thingies

Funfact for sorting junctions you use a cargo destination tagger to mark the sorting thing for the tagger. You can add in custom destinations to the tagger by screwdriver-->multitooling the tagger to unlock it allowing you to add/delete stuff to it.

**Comment by kol1th** [comment]
Delete this pr

**Comment by jakmak** [comment]
Kolith what exactly do you not like about it. If we get your input we could workshop it and maybe implement it into this PR

**Comment by PJB3005** [comment]
> Funfact for sorting junctions you use a cargo destination tagger to mark the sorting thing for the tagger. You can add in custom destinations to the tagger by screwdriver-->multitooling the tagger to unlock it allowing you to add/delete stuff to it.

Wow somebody other than me (I coded it) knows about this! Amazing!

**Comment by Kurfursten** [comment]
I knew about this too, but you can thank Taxi for that. The medbay station's package sorter had a destination tagger with an open cover.

**Comment by PJB3005** [comment]
That's because I mapped that into taxi in the PR I added it.

**Comment by AutismSpider** [comment]
only thing that needs a delay is disposal bins outlets inlets etc this can be use to block hallways etc

**Comment by Lokiamis** [comment]
>all it does is make construction unnecessarily tedious and slows down autism projects.

I don't have a problem with construction being tedious or slow. Damage to the station should be a hassle to fix - and this is what things should be balanced around, not autism projects. Some things don't have build times, but piping isn't a rack or generic locker, it's infrastructure.

**Comment by AutismSpider** [comment]
Get real @Lokiamis only people who give shit about the pipes or any type of building work are mommis and people with massive autism.

**Comment by jakmak5** [comment]
@autismspider already taken care of disposal pipes still have the delay while regular pipes are delay free

and @Lokiamis the delay is not noticeable for a 5 pipe fix from a welderbomb. If you add 2 seconds to each pipe than for 5 pipes it takes 10 seconds to construct. For autism projects like TEG's, SME, and the mythical R-UST you need a immense amount of pipe work. There exists no reason for this delay to exist and having it does not add to any other form of "balance" and only punishes engineers/mommi's.

**Comment by PJB3005** [comment]
How about upgraded RPD from science

**Comment by Aurx** [comment]
>another R&D toy that never gets made except by power gay scientists

No thanks.

**Comment by BobdaBiscuit** [comment]
I would go to science to order one as engi/atmos tech.

**Comment by AutismSpider** [comment]
@PJB3005 more shit that never get made unless you get into slap fight with sci fucking hell you need yell at sci to get adv kits and blue space beakers made for fucking medbay.

**Comment by jakmak5** [comment]
@PJB3005 if its a item that needs materials then you can bet your ass it would never be made. Even if metal/glass only the few people you would see make it would be the ones that broke into sci to get it. There is no reason to put efficient pipe autism past a time gate since it balances nothing any only hurts dedicated engineers/mommis

**Comment by PJB3005** [comment]
Power creep. 

What's next, 5 layer mode? Multiple directions at once?

There used to be a time when layers weren't a thing and people did fine. Cluster even told me to make unaries and such not be stackable anymore and I never got around to it, which I sort of regret now.

We shouldn't buff the RPD like this. If it needs to be more efficient to use we should make macros possible and make it easier to use efficiently.

**Comment by Kurfursten** [comment]
I don't have strong opinions about having an upgraded RPD in science, but the mineral gate does not stop things from getting made by any means. I still pick up my somatoray every shift as a botanist without fail. I can actually get my somatoray more reliably than I get get cargo to send me my ranching crate.

**Comment by AutismSpider** [comment]
@PJB3005 talk about power creep i point out engi borg material synthesizer human one sucks donkey dick to use and mommi one just for building stuff.

**Comment by Kurfursten** [comment]
The existence of powercreep elsewhere is not and has never been a good argument for more powercreep.

**Comment by jakmak5** [comment]
@pjb3005 i was litterally going to attempt 5 layer mode after this. For powercreeping something like deconstructing R walls to add a special tool to make it easier it is a major balance issue since R walls are intented to slow an intruder massively. It affects Malf AI's and securites ability to defend the armory. It would be a serious powercreep issue. For something as simple as making it easier to vend pipes does not affect anyone in a negative way. It does not affect antags or security in a noticable way. It is easy as hell to generate the minimum 150000 watts of power to fuel the station for the whole round and your super autistic 500 trillion watt sme will likely not affect the round or the antags in a noticeable way. The only people this affects are good engineers and mommis who sit in their autism forts tinkering their pipework and trying to see how much power they can generate. The RPD is a tool used exclusively for autism engineering and it does not serve a weaponized purpose.

**Comment by PJB3005** [comment]
The delay is the least bad part of the RPD. Removing the delay won't do Shit to make it better.

5 layer mode is literally over my dead body.

**Comment by SonixApache** [comment]
Reducing the time it takes to muy rapido effect atmos is pretty easily weaponised, y'know.
If you can just shit out sets of 5 stacked scrubbers in a room you can undo flooding pretty easily, and vice versa by shitting out vents.
Not to mention over/underpressurising, superheat/supercooling the air supply, etc...

**Comment by AutismSpider** [comment]
@SonixApache  people rarely try fix atmos damage even you know this

**Comment by AutismSpider** [comment]
@Kurfursten you are one to talk about power creep that pos wheelchair should have never seen light of day only weakness it has can be counter by hulking out is easy to do.

**Comment by Kurfursten** [comment]
You're a broken record, aren't you? Scolded once for using powercreep as a justification, you immediately try it again.

Well, that's not fair to the record. It could probably type better than you.

**Comment by PJB3005** [comment]
@AutismSpider take a chill pill Jesus.

**Comment by AutismSpider** [comment]
@PJB3005 no  want talk about power creep and am listing shit that is power creep

**Comment by PJB3005** [comment]
Are you unable to understand how conversations work?

**Comment by Exxion** [comment]
There was no power creep involved with the Medical Malpractice.
It *started out* OP

**Comment by Kurfursten** [comment]
True, if anything it creeped down in power.

Maybe this is where Bumbling Idiot reveals it was all a grand analogy and all along he actually meant that he wants to creep down power on the RPD, returning to ye old pipe dispenser!

**Comment by PJB3005** [comment]
FYI I'm actually working on mouse wheel support for the RPD, using the following controls:
* scroll: change dir 
* shift+scroll: change layer
* ctrl+scroll: change pipe based on a selectable list of favorites.

**Comment by Probe1** [comment]
I want no part in any of this PR or even the ones suggested.

We already have one of the easiest to use pipe systems.  I don't understand why it should be made even easier.  I'm yet to hear any discussion about why that needs to happen, other than "We don't like waiting".

Does it make atmos unplayable?
Does it make atmos really unenjoyable?
Does atmos suffer from this?
Are there balance issues created by it?
Is it adding a new quality of life change?

**Comment by PJB3005** [comment]
RPD is really painful to use due to the insane amount of mouse movement to do anything. It feels like Shit especially with a Shit mouse like me.

We shouldn't balance it around that. I'd be more than down for removing stacking if it means better RPD qol.

**Comment by Exxion** [comment]
It wouldn't
You can just choose not to stack right now

**Comment by PJB3005** [comment]
@Exxion see https://github.com/d3athrow/vgstation13/pull/4953#issuecomment-113847302

**Comment by N3X15** [comment]
6:+1: 12:-1:

**Pull request closed without merging**